src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ADRController.kt

```diff
@@ -167,20 +167,19 @@ class ADRController(private val encryption: Encryption,
     }
 
     @PostMapping("/datasets/{id}/resource/{resourceType}/{modelCalibrateId}")
-    @Suppress("ReturnCount")
+    @Suppress("ReturnCount", "LongParameterList")
     fun pushFileToADR(@PathVariable id: String,
                       @PathVariable resourceType: String,
                       @PathVariable modelCalibrateId: String,
                       @RequestParam resourceFileName: String,
-                      @RequestParam resourceId: String?): ResponseEntity<String>
+                      @RequestParam resourceId: String?,
+                      @RequestParam description: String) : ResponseEntity<String>
     {
         // 1. Download relevant artefact from hintr
         val artefact = when (resourceType)
         {
-            appProperties.adrOutputZipSchema -> Pair(apiClient.downloadSpectrum(modelCalibrateId),
-                    "Naomi model outputs")
-            appProperties.adrOutputSummarySchema -> Pair(apiClient.downloadSummary(modelCalibrateId),
-                    "Naomi summary report")
+            appProperties.adrOutputZipSchema -> Pair(apiClient.downloadSpectrum(modelCalibrateId), description)
+            appProperties.adrOutputSummarySchema -> Pair(apiClient.downloadSummary(modelCalibrateId), description)
```

Comment:
I think you can get rid of these `Pair`s - they're an unnecessary level of indirection if description is in common:
```suggestion
            appProperties.adrOutputZipSchema -> apiClient.downloadSpectrum(modelCalibrateId)
            appProperties.adrOutputSummarySchema -> apiClient.downloadSummary(modelCalibrateId)
```